crates/proto/src/serialize/binary/encoder.rs

```diff
@@ -439,6 +486,33 @@ impl<'a> BinEncoder<'a> {
     }
 }
 
+/// Determines whether name compression inside RDATA fields.
+#[derive(Clone, Copy)]
+pub enum RdataCompression {
+    /// Compression is allowed inside RDATA.
+    ///
+    /// This is only applicable to well-known record types defined in RFC 1035.
+    CompressionAllowed,
```

Comment:
Nit: don't repeat the same word between the type name and the variant, this is known as "stuttering". Maybe `{ Allowed, None }`?